source/loader/layers/sanitizer/asan_interceptor.cpp

```diff
@@ -708,8 +708,7 @@ ur_result_t SanitizerInterceptor::prepareLaunch(
         };
 
         // Write debug
-        // We use "uint64_t" here because EnqueueWriteGlobal will fail when it's "uint32_t"
```

Comment:
I didn't observe this fail when running locally, and I discovered that changing this to `uint32_t` fixes a test for the Debug option. Was this a fix for a specific platform @AllanZyne @yingcong-wu?

Reply:
A fix related to this var is purposed here: https://github.com/oneapi-src/unified-runtime/pull/1878

Reply:
> I didn't observe this fail when running locally, and I discovered that changing this to `uint32_t` fixes a test for the Debug option. Was this a fix for a specific platform @AllanZyne @yingcong-wu?

I remember correctly, `uint32_t` was not worked on PVC. 
If you tested this and it worked, I also think it'd be better to use `uint32_t`.

Reply:
PR #1878 is merged recently, you can see if that fix the problem you are trying to fix with this change.